src/dbnode/storage/shard.go

```diff
@@ -967,23 +967,27 @@ func (s *dbShard) newShardEntry(
 	clonedID.NoFinalize()
 
 	var (
-		clonedTags ident.Tags
-		err        error
+		tags ident.Tags
+		err  error
 	)
-	if tags.Remaining() > 0 {
-		clonedTags, err = convert.TagsFromTagsIter(
-			clonedID, tags.Duplicate(), s.identifierPool)
-		tags.Close()
+
+	if tagsIter.Remaining() > 0 {
+		dupTagsIter := tagsIter.Duplicate()
```

Comment:
You need to duplicate before doing the `Remaining()` check because it may be at the end of its iterating and have `Remaining() == 0` in that case (before duplication)